### ISSUE TITLE
Workspace inheritance is broken

### ISSUE THREAD

**Issue by ederst** [kbst/terraform-kubestack#176]
**What happened?**

After the upgrade from `v0.12.2-beta.0` to `v0.13.0-beta.0` the inheritance between the workspaces is broken. Specified variables in `config.auto.tfvars` in `ops` and `loc` are ignored by the deployment.

However, it is possible to work around the issue by specifying the variable in the base (`apps`) workspace.

**What is the expected behavior?**

The expected behavior is (probably) not needing to set the variable in the configuration of the base workspace, and the variable being used anyways.

**How to reproduce the issue?**

To reproduce the issue, set a variable in any workspace inheriting from `apps`, for example:

```hcl
# loc environment, inherits from apps
loc = {
  node_image = "ghcr.io/kbst/kind-eks-d:v1.18.9-kbst.1_loc"
}
```

Switch to the workspace and plan/apply:

```shell
❯ terraform workspace select loc    
Switched to workspace "loc".

❯ terraform plan -no-color > plan.txt
```

Observe the output:

```text
  # module.eks_zero.module.cluster.kind_cluster.current will be created
  + resource "kind_cluster" "current" {
      # ... yadda, yadda, yadda
      + name                   = "stacktest-loc-eks-e3b0c44"
      + node_image             = (known after apply)
```

Now set the variable in the `apps` configuration:

```hcl
    apps = {
      # much config, such wow
      node_image = ""
    }
```

Then run plan/apply again and have a look at the output again:

```text
  + resource "kind_cluster" "current" {
      # amazing changes here, but the next line is important:
      + node_image             = "ghcr.io/kbst/kind-eks-d:v1.18.9-kbst.1_loc"
```

**Additional information**

The error might be located somewhere here (at least this was some significant change I found comparing v0.12 vs v0.13, but of course I could be wrong):

https://github.com/kbst/terraform-kubestack/blob/351e51388240f71e2a2a45a24f3a5b0c3af3bfd4/common/configuration/outputs.tf#L6-L18

Edit: I am not wrong, the code from L13-14 only accounts for the keys in the base config and ignores the ones in the child environments.

**Information about the environment**:

```hcl
module "eks_zero" {
  providers = {
    aws = aws.eks_zero
  }

  source = "github.com/kbst/terraform-kubestack//aws/cluster?ref=v0.13.0-beta.0"
  # source = "github.com/kbst/terraform-kubestack//aws/cluster-local?ref=v0.13.0-beta.0"

  configuration = var.clusters["eks_zero"]
}
```

**Comment by pst** [comment]
I just hit the exact same issue myself. Working on a fix as we speak.

**Comment by ederst** [comment]
@pst I tried to fix it myself, just for fun of it: https://github.com/ederst/terraform-kubestack/commit/293416af8aeace2f870389b06ecd75d30e339234

**Comment by pst** [comment]
Nice, I actually think your use of setunion is cleaner than my toset and concat. Happy to test and merge your fix if you send a PR.

**Comment by ederst** [comment]
There you go, i stole your commit message and explanation though, because it IMHO better than mine.

**Comment by ederst** [comment]
I wonder if it is possible to add tests for this as well. Will have a look at it.

**Comment by pst** [comment]
Yeah unit tests for modules like that would really help. I am experimenting using the Terraform plugin SDKs testing for that. But it's still just an experiment.

**Comment by ederst** [comment]
Have you tried this: https://github.com/gruntwork-io/terratest/

And there seems to be talk about a [`terraform test` command](https://www.hashicorp.com/blog/testing-hashicorp-terraform). By the looks of it this will be available with [0.15.x](https://github.com/hashicorp/terraform/blob/v0.15.0-rc2/command/arguments/test.go). So might be something to wait for.

**Comment by pst** [comment]
Released v0.13.0-beta.1 with fix https://github.com/kbst/terraform-kubestack/releases/tag/v0.13.0-beta.1
